src/stream/src/executor/over_window/over_partition.rs

```diff
@@ -760,577 +914,3 @@ impl<'a, S: StateStore> OverPartition<'a, S> {
-                        cursor.move_prev();
-                        if cursor.position().is_ghost() {
-                            break;
-                        }
-                    }
-                    cursor.key().unwrap_or(first_key)
-                }
-            };
-            min_frame_start = min_frame_start.min(key);
-            if min_frame_start == first_key {
-                // if we already pushed the affected frame start to the first key, no more pushing is needed
-                break;
-            }
-        }
-
-        min_frame_start
-    };
-
-    let last_curr_key = if start_is_unbounded || delta_last_key == last_key {
-        last_key
-    } else {
-        let mut max_last_curr_key = &Sentinelled::Smallest;
-
-        for call in calls {
-            let key = match &call.frame.bounds {
-                FrameBounds::Rows(bounds) => {
-                    let mut cursor = part_with_delta.upper_bound(Bound::Included(delta_last_key));
-                    for _ in 0..bounds.start.n_preceding_rows().unwrap() {
-                        cursor.move_next();
-                        if cursor.position().is_ghost() {
-                            break;
-                        }
-                    }
-                    cursor.key().unwrap_or(last_key)
-                }
-            };
-            max_last_curr_key = max_last_curr_key.max(key);
-            if max_last_curr_key == last_key {
-                // if we already pushed the affected curr key to the last key, no more pushing is needed
-                break;
-            }
-        }
-
-        max_last_curr_key
-    };
-
-    let last_frame_end = if end_is_unbounded || last_curr_key == last_key {
-        last_key
-    } else {
-        let mut max_frame_end = &Sentinelled::Smallest;
-
-        for call in calls {
-            let key = match &call.frame.bounds {
-                FrameBounds::Rows(bounds) => {
-                    let mut cursor = part_with_delta.find(last_curr_key).unwrap();
-                    for _ in 0..bounds.end.n_following_rows().unwrap() {
-                        cursor.move_next();
-                        if cursor.position().is_ghost() {
-                            break;
-                        }
-                    }
-                    cursor.key().unwrap_or(last_key)
-                }
-            };
-            max_frame_end = max_frame_end.max(key);
-            if max_frame_end == last_key {
-                // if we already pushed the affected frame end to the last key, no more pushing is needed
-                break;
-            }
-        }
-
-        max_frame_end
-    };
-
-    if first_curr_key > last_curr_key {
-        // all affected keys are deleted in the delta
-        return vec![];
-    }
-
-    vec![(
-        first_frame_start,
-        first_curr_key,
-        last_curr_key,
-        last_frame_end,
-    )]
-}
-
-#[cfg(test)]
-mod find_affected_ranges_tests {
```

Comment:
Are we able to repurpose some of these old tests for the new implementation?

Reply:
Yes, will migrate some old tests to test `find_first_curr_for_rows_frame` etc